lib/jekyll/tags/highlight.rb

```diff
@@ -39,6 +40,17 @@ def initialize(tag_name, markup, tokens)
         end
       end
 
+      def parse(tokens)
+        if @lang == 'raw'
+          end_pos = tokens.rindex{|x| x =~ END_HIGHLIGHT }
```

Comment:
> Why would you shift to get an index?

I mean normally we don't need to scan the list of tokens, we just get and consume the next token.